Sources/AblyAssetTrackingInternal/WorkerQueue/Worker.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+import AblyAssetTrackingCore
+
+/// A [Worker] interface represents workers which execute work inside [WorkerQueue].
+/// PropertiesType: - the type of properties used by this worker as both input and output.
+/// WorkerSpecificationType - the type of specification used to post another worker back to the queue.
+public protocol Worker<PropertiesType, WorkerSpecificationType>: AnyObject {
+    associatedtype PropertiesType
+    associatedtype WorkerSpecificationType
+    
+    var completion: ResultHandler<Void> { get }
+    var workerTypeDescription: String { get }
+    
+    /// This function is provided in order for implementors to implement synchronous work. Any asynchronous tasks
+    /// should be executed inside [doAsyncWork] function. If a worker needs to delegate another task to the queue
+    /// pass it to [postWork] function.
+    /// - parameters:
+    ///    - properties: current state of publisher to be used by this worker.
+    ///    - doAsyncWork: wrapper function for asynchronous work.
+    ///    - postWork: function that allows worker to add other workers to the queue calling it.
+    /// - Returns: updated Properties modified by this worker.
+    func doWork(
+        properties: PropertiesType,
+        doAsyncWork: (() throws -> Void) -> Void,
```

Comment:
This seems to be performing synchronous work, not asynchronous. Contrast it with Android, where this closure [takes a suspending function as its parameter](https://github.com/ably/ably-asset-tracking-android/blob/1a1869ef205280ca520e97f28d0b2733231f372e/common/src/main/java/com/ably/tracking/common/workerqueue/Worker.kt#L25).

[From the documentation written by Kacper](https://github.com/ably/ably-asset-tracking-common/tree/main/specification#running-asynchronous-operations-from-the-synchronous-event-queue), the idea of `doAsyncWork` is to not "block the queue to wait for async operations to complete because we can be doing other work during that time", so our worker queue needs to expose a similar mechanism to allow work to be run in the background without blocking the worker queue (perhaps a separate concurrent queue?)

Can we also please add documentation to the code explaining the asynchronous work mechanism?

Reply:
... good catch. I'll need to think about how to approach this one.

Reply:
I notice that you haven't changed the signature of the `doAsyncWork` parameter. It still accepts a `() -> Void` closure. To be consistent with the suspending function accepted in Android, shouldn't this be a closure that accepts a completion handler, i.e. `(() -> Void) -> Void`?

Reply:
doAsyncWork is specified as `(@escaping () throws -> Void) -> Void` from what I'm seeing. Isn't that exactly what you're suggesting? Sorry, maybe I'm missing something obvious 😅 
```
        doAsyncWork: (@escaping () throws -> Void) -> Void,
```

Reply:
I'm thinking it'd be

```
doAsyncWork: (an @escaping function that completes via a callback) -> Void
```

i.e.

```
doAsyncWork: (@escaping (a completion handler that can receive an error) -> Void)
```

i.e.

```
doAsyncWork: (@escaping ((Error?) -> Void) -> Void)
```

— does that make sense?

Reply:
Yeah I think I see the issue now, had to look into how kotlin's suspending functions work again. 
ca3fee07fd0733f64b5dbec9b5f6632f6fc20714

Reply:
I have a few questions about this (and the async work in general) - as it is subtly different from Android, lets sync after our planning meeting today :)